src/routes/safe/components/Transactions/helpers/TxModalWrapper/index.tsx

```diff
@@ -109,9 +110,11 @@ export const TxModalWrapper = ({
   const showCheckbox = !isSpendingLimitTx && canTxExecute && (!txThreshold || txThreshold > confirmationsLen)
   const nativeCurrency = getNativeCurrency()
   const { currentVersion: safeVersion, threshold } = useSelector(currentSafe) ?? {}
-  const { smartContractWallet } = useSelector(providerSelector)
   const isCreation = isMultisigCreation(confirmationsLen, txType)
-  const isOffChainSignature = checkIfOffChainSignatureIsPossible(doExecute, smartContractWallet, safeVersion)
+  const userAccount = useSelector(userAccountSelector)
+  const isSmart = useIsSmartContract(userAccount)
```

Comment:
No need to pass `userAccount` as an argument, the hook itself could just call the selector.

Reply:
Why? It can be used much like the function in this way.

Reply:
I mean, in this particular case userAccount isn't used on its own. So it would be nice to encapsulate all the dependencies of the hook in the hook itself.

Reply:
I overlooked `userAddress` which is used elsewhere in the wrapper so I've switched to using that. As such, I think we should keep the argument.